lib/matplotlib/sphinxext/plot_directive.py

```diff
@@ -176,9 +177,10 @@
 
 def plot_directive(name, arguments, options, content, lineno,
                    content_offset, block_text, state, state_machine):
-    """Implementation of the ``.. plot::`` directive.
+    """Deprecated function-based implementation of the ``.. plot::`` directive.
```

Comment:
Use `@cbook.deprecated("3.1", alternative="PlotDirective")` to mark deprecation rather than editing the docstring (the decorator will edit it for you).